pages/products/[slug].tsx

```diff
@@ -235,10 +268,55 @@ const ProductPage = ({
               >
                 {loadingAddToCheckout ? "Adding..." : "Add to cart"}
               </button>
-            ) : (
+            ) : null}
+
+            {!selectedVariant && (
```

Comment:
To avoid big shifts in the UI when selecting the variant, lets just disable "add to cart" button when variant is not selected

Reply:
Button should be disabled also when variant is sold out. 

lets create new variable:
```
const isAddToCartButtonDisabled = !selectedVariant || selectedVariant?.quantityAvailable === 0
```

This way logic of disabling the button will be easier to follow and moved out of visual (rendering)